### PR TITLE
trigger system and new ui

### PR DESCRIPTION
**Pull request by srmcconomy**

### CODE REVIEW

File: src/server/models/Player.js [hunk 1]
```diff
@@ -86,33 +130,119 @@ export default class Player extends Model {
...
+      ].forEach(({ title, getAllCards }) => {
+        cards.optional[title] = getAllCards().filter(card => !handledCards.has(card) && card.canTriggerOn(event, this, title === 'persistent' ? 'persistent' : 'normal'));
+        cards.mandatory[title] = getAllCards().filter(card => !handledCards.has(card) && card.willTriggerOn(event, this, title === 'persistent' ? 'persistent' : 'normal'));
+      });
+      optionalCardsLength = Object.values(cards.optional).reduce((sum, list) => sum + list.length, 0);
+      mandatoryCardsLength = Object.values(cards.mandatory).reduce((sum, list) => sum + list.length, 0);
+    };
+    refreshCards();
+
+    while (optionalCardsLength > 0 || mandatoryCardsLength > 1) {
```
Comment (generalpepper) [inline src/server/models/Player.js:217]: probably need mandatoryCardsLength > 0 instead of 1

File: src/server/cards/seaside/outpost.js [hunk 2]
```diff
@@ -4,13 +4,32 @@ export default class Outpost extends Card {
...
   static types = new Set(['Action', 'Duration']);
   async onPlay(player) {
-    this.ignoreCleanUp = true;
-    player.playArea.filter(c => c !== this).forEach(c => {
-      if (c.title === 'Outpost') this.ignoreCleanUp = false;
-    });
-    if (player.game.previousPlayer === player) this.ignoreCleanUp = false;
+    if (
+      !player.cardsPlayedThisTurn.some(card => card !== this && card.title === 'Outpost') &&
+      !player.playArea.some(card => card !== this && card.title === 'Outpost')
```
Comment (generalpepper) [inline src/server/cards/seaside/outpost.js:9]: Still needs the check that the previous player wasn't the same player.

File: src/server/models/Player.js (hunk 1 above)
Comment (srmcconomy) [reply src/server/models/Player.js:217]: no, you don't get the option to select the last mandatory card
it just happens automatically

Comment (srmcconomy) [reply src/server/models/Player.js:217]: this way you don't have to click something to make a duration trigger when there is only one there

File: src/server/cards/seaside/outpost.js (hunk 2 above)
Comment (srmcconomy) [reply src/server/cards/seaside/outpost.js:9]: is there a way to remove outpost from the play area on the 2nd turn? if not, isnt it good enough to make sure that there are no outposts in the play area?

File: src/server/models/Player.js (hunk 1 above)
Comment (generalpepper) [reply src/server/models/Player.js:217]: That makes sense, forgot about the automatic implementation

File: src/server/cards/seaside/outpost.js (hunk 2 above)
Comment (generalpepper) [reply src/server/cards/seaside/outpost.js:9]: No there isn't and that would work. Except for Mission an Possession will mess with us. You can't outpost on a mission turn or after someone possessed you last turn.

Not sure if we ever make possession but should still plan for it's possibility.

Comment (srmcconomy) [reply src/server/cards/seaside/outpost.js:9]: ok will fix later

**Pull request merged**